fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java

```diff
@@ -797,99 +917,190 @@ protected void shutdown() {
+            // Optionally apply a read-based workload to stress the FHIR server and database
+            // with random requests for resources
+            if (this.concurrentPayerRequests > 0) {
+                if (fhirClient == null) {
+                    throw new IllegalArgumentException("Interop test workload requires FHIR client configuration");
+                }
+                
+                // The interop workload uses patient resource ids captured during previous data load runs
+                // and stored in the FHIRBUCKET.LOGICAL_RESOURCES table
+                InteropScenario scenario = new InteropScenario(this.fhirClient);
+                interopWorkload = new InteropWorkload(dataAccess, scenario, concurrentPayerRequests, this.patientBufferSize, this.bufferRecycleCount);
+                interopWorkload.init();
             }
         }
+    }
 
-        // Optionally apply a read-based workload to stress the FHIR server and database
-        // with random requests for resources
-        if (this.concurrentPayerRequests > 0 && fhirClient != null) {
-            // set up the CMS payer thread to add some read-load to the system
-            InteropScenario scenario = new InteropScenario(this.fhirClient);
-            cmsPayerWorkload = new InteropWorkload(dataAccess, scenario, concurrentPayerRequests, this.patientBufferSize, this.bufferRecycleCount);
-            cmsPayerWorkload.init();
+    /**
+     * Use fhir-bucket to drive the (parallel) reindex process.
+     */
+    private void doReindex() {
+        if (this.clientSideDrivenReindex) {
+            this.driveReindexOperation = new ClientDrivenReindexOperation(fhirClient, reindexConcurrentRequests, reindexTstampParam, reindexResourceCount, reindexStartWithIndexId);
+        } else {
+            this.driveReindexOperation = new ServerDrivenReindexOperation(fhirClient, reindexConcurrentRequests, reindexTstampParam, reindexResourceCount);
         }
+        this.driveReindexOperation.init();
+        
+    }
 
-        // Optionally start the $reindex loops
-        if (this.reindexTstampParam != null) {
-            if (this.clientSideDrivenReindex) {
-                this.driveReindexOperation = new ClientDrivenReindexOperation(fhirClient, reindexConcurrentRequests, reindexTstampParam, reindexResourceCount, reindexStartWithIndexId);
-            } else {
-                this.driveReindexOperation = new ServerDrivenReindexOperation(fhirClient, reindexConcurrentRequests, reindexTstampParam, reindexResourceCount);
-            }
-            this.driveReindexOperation.init();
+    /**
+     * Start a scanner to find resources to read and load
+     * @param dataAccess
+     */
+    private void startScanner(DataAccess dataAccess) {
+        if (baseDirectory != null) {
+            // Use a local file scanner, instead of a COS scanner
+            // Data may be null here, which is OK because we're not using the database
+            // to track anything
```

Comment:
is there a local file mode where the db *is* used?
for some reason I was thinking LocalFile was for that, whereas ImmediateLocalFile was for the no-db case...

Reply:
The --immediate-local option enables the no-db immediate loading case. I added that initially to make it easy to simply load the synthea data, but after that I wanted to record the new logical ids so that I could drive the interop workload locally and for that I need to register the local files during loading (because of how the schema ties things together). So I added a second option to scan files and register them in the database. So now we have a lot of options. I'll update the README.